core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```diff
@@ -162,6 +161,25 @@ public ClusterState execute(ClusterState currentState) throws Exception {
         });
     }
 
+    void validatePipeline(Map<DiscoveryNode, IngestInfo> ingestInfos, PutPipelineRequest request) throws Exception {
```

Comment:
what is this validation for mainly? Is it to make sure that if you have an ingest plugin, it got installed on all nodes? All the builtin processors are guaranteed to be on every node with 5.0 right?